.github/workflows/build_wheels_macos.yml

```diff
@@ -0,0 +1,119 @@
+name: Build MacOS wheels
+
+on:
+  push:
+    branches:
+      - main # just build the sdist & wheel, skip release
+    tags:
+      - "*"
+  pull_request: # also build on PRs touching this file
+    paths:
+      - ".github/workflows/build_wheels_macos.yml"
+      - "ci/*"
+      - "MANIFEST.in"
+      - "pyproject.toml"
+      - "setup.py"
+  workflow_dispatch:
+
+jobs:
+  build_wheels:
+    name: Build wheels on ${{ matrix.os }} ${{ matrix.platform }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: ["macos-latest"]
+        platform: ["x86_64"]
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Cache vcpkg
+        uses: actions/cache@v1
+        id: vcpkgcache
+        with:
+          path: "/Users/runner/.cache/vcpkg/archives"
```

Comment:
In the Windows PR, I am caching the actual "/installed" directory directly (https://github.com/geopandas/pyogrio/pull/49/files#diff-6c1677623c2a63cdf28017fb8aa9b0395ecf72745a91b9ccd9228dbac36b4475R34). Do you know if there is a big difference? (both seem to work, and I think I have seen both in practice in other projects)

Reply:
I went back to `/Users/runner/.cache/vcpkg/archives` only because it was getting emitted by the logs on GH, and I couldn't figure out the correct install location.

For me, I had git cloned vcpkg to a local folder, so the `installed/x64-osx-dynamic/` path ended up being under that.

From earlier testing of just building GDAL, it seems like it cached things fine using `/Users/runner/.cache/vcpkg/archives` and may be preferable to use that if that is based on a hash relating to the version of GDAL + deps that are actually installed, if we want those to rebuild on updates from vcpkg (which is debatable).